packages/chain-helpers/src/errorhandling/ErrorHandler.ts

```diff
@@ -72,20 +57,16 @@ export function extrinsicSuccessful(
  */
 export function getExtrinsicError(
   extrinsicResult: ISubmittableResult
-): ExtrinsicError | null {
+): RegistryError | DispatchError | null {
   const errorEvent = extrinsicResult.dispatchError
 
   if (errorEvent && errorEvent.isModule) {
     const moduleError = errorEvent.asModule
-
-    const index = moduleError.index.toNumber()
-    const error = moduleError.error.toNumber()
-    if (index >= 0 && error >= 0) {
-      return errorForPallet({ index, error })
+    try {
+      return moduleError.registry.findMetaError(moduleError)
+    } catch {
+      // handled below
```

Comment:
Maybe different wording, in case more code comes in between in the future?
Like "handled with last return"?